src/mediapipe_calculators/python_executor_calculator.cc

```diff
@@ -0,0 +1,160 @@
+        RET_CHECK(!cc->Inputs().GetTags().empty());
+        RET_CHECK(!cc->Outputs().GetTags().empty());
+        for (auto& input : cc->Inputs()) {
+            input.Set<PyObjectWrapper<py::object>>();
+        }
+        for (auto& output : cc->Outputs()) {
+            output.Set<PyObjectWrapper<py::object>>();
+        }
+        cc->InputSidePackets().Tag(INPUT_SIDE_PACKET_TAG).Set<PythonNodesResources>();
+        LOG(INFO) << "PythonExecutorCalculator [Node: " << cc->GetNodeName() << "] GetContract end";
+        return absl::OkStatus();
+    }
+
+    absl::Status Close(CalculatorContext* cc) final {
+        LOG(INFO) << "PythonExecutorCalculator [Node: " << cc->NodeName() << "] Close";
+        return absl::OkStatus();
+    }
+
+    absl::Status Open(CalculatorContext* cc) final {
+        LOG(INFO) << "PythonExecutorCalculator [Node: " << cc->NodeName() << "] Open start";
+        PythonNodesResources nodesResourcesPtr = cc->InputSidePackets().Tag(INPUT_SIDE_PACKET_TAG).Get<PythonNodesResources>();
+        auto it = nodesResourcesPtr.find(cc->NodeName());
+        if (it == nodesResourcesPtr.end()) {
+            LOG(INFO) << "Could not find initialized Python node named: " << cc->NodeName();
+            RET_CHECK(false);
+        }
+        nodeResources = it->second;
+        LOG(INFO) << "PythonExecutorCalculator [Node: " << cc->NodeName() << "] Open end";
+        return absl::OkStatus();
+    }
+
+    absl::Status Process(CalculatorContext* cc) final {
+        LOG(INFO) << "PythonExecutorCalculator [Node: " << cc->NodeName() << "] Process start";
+        py::gil_scoped_acquire acquire;
+        try {
+            py::print("PYTHON: Acquired GIL");
+            if (pyIteratorPtr) { // Iterator initialized
+                py::print("PyIterator initialized block");
+                if (pyIteratorPtr->getImmutableObject() != py::iterator::sentinel()) {
+                    py::list pyOutputs = py::cast<py::list>(*pyIteratorPtr->getImmutableObject());
+                    pushOutputs(cc, pyOutputs);
+                    ++(pyIteratorPtr->getMutableObject());
+                }
+            } else { // Iterator not initialized, either first iteration or execute is not yielding
+                py::print("PyIterator uninitialized block");
+                std::vector<py::object> pyInputs;
+                for (const std::string& tag : cc->Inputs().GetTags()) {
+                    const py::object& pyInput = cc->Inputs().Tag(tag).Get<PyObjectWrapper<py::object>>().getImmutableObject();
+                    nodeResources->pythonBackend->validateOvmsPyTensor(pyInput);
+                    pyInputs.push_back(pyInput);
+                }
+            
+                py::object executeResult = std::move(nodeResources->nodeResourceObject->attr("execute")(pyInputs));
+                py::print(executeResult.attr("__class__").attr("__name__"));
```

Comment:
Why do we need to use py::print? There already is one logging mechanism for calculators - LOG(DEBUG)

Reply:
It's a debug code, will be removed in next commits.